stackslib/src/net/api/gettenureblocks.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (C) 2025 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use clarity::types::chainstate::StacksBlockId;
+use regex::{Captures, Regex};
+use serde_json;
+use stacks_common::types::chainstate::{BlockHeaderHash, ConsensusHash};
+use stacks_common::types::net::PeerHost;
+
+use crate::chainstate::nakamoto::NakamotoChainState;
+use crate::chainstate::stacks::db::{StacksBlockHeaderTypes, StacksHeaderInfo};
+use crate::net::http::{
+    parse_json, Error, HttpNotFound, HttpRequest, HttpRequestContents, HttpRequestPreamble,
+    HttpResponse, HttpResponseContents, HttpResponsePayload, HttpResponsePreamble, HttpServerError,
+};
+use crate::net::httpcore::{request, RPCRequestHandler, StacksHttpRequest, StacksHttpResponse};
+use crate::net::{Error as NetError, StacksNodeState};
+
+#[derive(Clone)]
+pub struct RPCNakamotoTenureBlocksRequestHandler {
+    pub(crate) consensus_hash: Option<ConsensusHash>,
+}
+
+impl RPCNakamotoTenureBlocksRequestHandler {
+    pub fn new() -> Self {
+        Self {
+            consensus_hash: None,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct RPCTenureBlock {
```

Comment:
You probably also want an epoch ID, or at least an indication as to whether or not this is a Stacks 2.x block or a Nakamoto block